### PR TITLE
DAOS-5123 REBUILD: Rebuild Protocol and Tests for OSA Drain.

### PR DESCRIPTION
**Pull request by PetFet**
This patch includes the changes necessary for rebuild to migrate data for OSA Drain.

- Several modifications for rebuild to facilitate the drain command/operation.
- Refactored the rebuild tests for easier extensibility.
- Implemented simple tests for Drain.

### CODE REVIEW

File: src/tests/suite/daos_test.c [hunk 1]
```diff
@@ -241,6 +242,13 @@ run_specified_tests(const char *tests, int rank, int size,
 			nr_failed += run_daos_rebuild_simple_test(rank, size,
 						sub_tests, sub_tests_size);
 			break;
+		case 'b':
```
Comment (NiuYawei) [inline src/tests/suite/daos_test.c:245]: The short option should be abbr. of long option, but seems lots of the short options in this file didn't follow the rule.

**Comment by PetFet** [comment]
@shimizukko 

Scott mentioned you might be able to take a quick look at this.

I've added a new test called drain to daos_test which uses the -b flag.
I've tested it locally but it seems as though it's not running in CI. I was hoping you might know what I was missing so that it gets picked up and run alongside the other tests.

**Comment by shimizukko** [comment]
Hi Peter.
CI uses Avocado to run daos_test, so you should add your test into:
https://github.com/daos-stack/daos/blob/master/src/tests/ftest/daos_test/daos_core_test.yaml
You would add something like:
```
test_b:
      daos_test: b
      test_name: Drain tests
      test_timeout: 1500
```

**Comment by PetFet** [comment]
Drain tests are now definitely running in CI and have passed. Should be ready for merge.

**Pull request merged**